runtime/js/src/main/kotlin/kotlinx/serialization/DynamicObjectParser.kt

```diff
@@ -56,6 +60,18 @@ class DynamicObjectParser(): AbstractSerialFormat() {
             }
         }
 
+        override fun decodeTaggedLong(tag: String): Long {
+            val obj = getByTag(tag)
+            val number = obj as? Double ?: throw SerializationException("$obj is not a Number")
+            val canBeConverted = number.isFinite() && floor(number) == number
+            if (!canBeConverted)
+                throw SerializationException("$number can't be represented as Long")
+            val inBound = abs(number) <= MAX_SAFE_INTEGER
+            if (!inBound)
+                throw SerializationException("$number does not fit in Long due to a potential precision loss")
```

Comment:
Maybe "does not fit" => "could not be deserialized"? Otherwise the message might be confusing...